### ISSUE TITLE
feat(COMPONENT): allow multiple instances of AriaDescriber

### ISSUE THREAD

**Issue by yann-armelin** [angular/components#23499]
#### Feature Description

Allow to have several [AriaDescriber](https://github.com/angular/components/blob/master/src/cdk/a11y/aria-describer/aria-describer.ts) containers on a same page.
Today the container of `AriaDescriber` is necessarily appended to body, with an hardcoded id, and any pre-existing container is removed.

I see At least 3 different options to achieve this:

1. Do something similar to what's done in [LiveAnnouncer](https://github.com/angular/components/blob/master/src/cdk/a11y/live-announcer/live-announcer.ts): use an Injection token to inject the container.

2. Make the following constant customizable: MESSAGES_CONTAINER_ID . So that each instance of `AriaDescriber` can have his own container.

3. Do something similar to [OverlayContainer](https://github.com/angular/components/blob/master/src/cdk/overlay/overlay-container.ts) so that the container can be moved somewhere else on the page, and stop relying on the ID the remove it 

In all cases, the following const should be customizable: `CDK_DESCRIBEDBY_ID_PREFIX`, otherwise elements with duplicated ID will be created on the page.

#### Use Case

In the context of micro frontends, we may end up with 2 Angular Material instances on the same page.

Today, a second instance of `AriaDescriber` would remove the container of the first one, making it ineffective.

**Comment by crisbeto** [comment]
Looking through it, I think that it should mostly work already. The `messagesContainer` is a global variable which means that even if you ended up with multiple instances of the `AriaDescriber`, they'll all still use the same container.

That being said, I think that this will break down if the an `AriaDescriber` is destroyed while another one is still live, because we remove the global container in `ngOnDestroy`. It'll also break down one `AriaDescriber` is initialized later than another one, because the descriptions of the former will be removed by the latter. I think that these are somewhat edge cases, but they can definitely happen.

I think the way to go is to do something similar to the `OverlayContainer`. I'll look into it later today.
